examples/third_party_examples/ai_agent_functions_node.md

```diff
@@ -0,0 +1,359 @@
+# Build an AI agent with OpenAI functions and the Node.js SDK
+
+OpenAI functions enable your app to take action based on user inputs. This means that it can, e.g., search the web, send emails, or book tickets on behalf of your users, making it more powerful than a regular chatbot.
+
+In this tutorial, you will build an app that uses OpenAI functions along with the latest version of the Node.js SDK. The app runs in the browser, so you only need a code editor and, e.g., VS Code Live Server to follow along locally. Alternatively, write your code directly in the browser via [this code playground.](https://scrimba.com/scrim/c6r3LkU9)
+
+If you prefer watching screencasts over reading, then you can check out [this scrim, which walks through the code line-by-line:](https://scrimba.com/scrim/co0044b2d9b9b7f5bf16e0391)
+
+<iframe src="https://scrimba.com/scrim/co0044b2d9b9b7f5bf16e0391?embed=openai,no-header" width="100%" height="500"></iframe>
+
+## What you will build
+
+Our app is a simple agent that helps you find activities in your area.
+It has access to two functions, `getLocation()` and `getCurrentWeather()`,
+which means it can figure out where you’re located and what the weather
+is at the moment. 
+
+At this point, it's important to understand that
+OpenAI doesn't execute any code for you. It just tells your app which
+functions it should use in a given scenario, and then leaves it up to
+your app to invoke them.
+
+Once our agent knows your location and the weather, it'll use GPT’s
+internal knowledge to suggest suitable local activities for you.
+
+## Importing the SDK and authenticating with OpenAI
+
+We start by importing the OpenAI SDK at the top of our JavaScript file and authenticate with our API key, which we have stored as an environment variable.
+
+``` js
+import OpenAI from "openai";
+
+const openai = new OpenAI({
+    apiKey: process.env.OPENAI_API_KEY,
+    dangerouslyAllowBrowser: true
+});
+```
+
+We also need to set `dangerouslyAllowBrowser: true`, as OpenAI won't let us send requests to their API from a browser unless we have confirmed that we are aware of the risks involved with that.
```

Comment:
This only applies to the scrimba environment right? Perhaps worth clarifying that all of this can also be run locally as a node script, and that we don't encourage this practice in a production setup.

Reply:
Good point, I just modified the paragraph to include a recommendation of moving it to Node in production.